### PR TITLE
OOP school library: preserve data

### PR DESCRIPTION
**Pull request by muneeb3123**
This PR includes:

- [ ] On exit from your app save all the data (books, people, and rentals) into JSON files. One per type of data you are storing and named accordingly (e.g. rentals.json)

- [ ] On startup of your app load all data from files. Remember to handle the case when one or more of the files are missing.

### CODE REVIEW

File: app.rb [hunk 1]
```diff
@@ -154,12 +159,13 @@ def create_rental_with_book_and_person(input_book, input_person)
     print 'Date in format YYYY-MM-DD:  '
     date = @user_interface.user_input
     if /^\d{4}-\d{2}-\d{2}$/.match(date)
+      rental = Rental.new(date)
       selected_book = @books[input_book]
       selected_person = @persons[input_person]
-      rental = Rental.new(date)
       rental.add_book(selected_book)
       rental.add_person(selected_person)
       @rentals << rental
```
Comment (nahnahdev) [inline app.rb:168]: - [x] Great job so far team. You have done an amazing job so far and your app is able to preserve data as expected :clap: 
- [x] However, I deleted all the current `JSON` files in your project to test if the part of the project requirement that mentioned that you need to remember to handle the case when one or more of the files are missing and ensure that the app works and shows no errors 👍 
- [x] The app did start as expected. and when some inputs were made to create a new `person` and `book`, the respective `JSON files `were recreated and stored the data without any errors :clap: However, in an attempt to create a new `rental`, the app threw an error and stopped working. :worried: 
- [x] As a result, data wasn't completed persisted in your app as expected so kindly fix this and ensure that your app works as expected even if any of the `JSON files` are missing 😉 
- Don't forget to also ensure that it is possible to `List all rentals for a given person's ID` :+1: 
- [x] If you look at the video, you'll notice that the `Create a rental` and `List all rentals for a given person's ID` features didn't work as expected :+1:  So kindly work on this so your app works as expected :100: 
- Video Reference
_**I took this video after deleting all the JSON files from the project 👍**_

[Screencast from 07-12-2023 04:30:02 PM.webm](https://github.com/muneeb3123/library-books/assets/102699958/085dfd2c-682b-4585-900a-c1778eb421d3)

- Project Requirement
image

**Review by nahnahdev — CHANGES_REQUESTED** [review]
## Project Status: CHANGES REQUIRED :recycle: 
Hi @muneeb3123  and @CarlosIgreda   , 

Great work so far. There are just a few changes you need to make and you’ll be on your way to the next project.


 ## Highlights 🔥 
-  Good use of GitFlow ✔️
- Linters well setup and passing ✔️
- Most project requirements met ✔️

 ## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._


Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me @NahnahAJ in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: app.rb (hunk 1 above)
Comment (muneeb3123) [reply app.rb:168]: thanks brother i fixed it now all functions work

**Review by uwadonat — APPROVED** [review]
## Hi @Team,

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### Congratulations! 🎉 
### Highlights
- No linter errors ✔️
- Detailed pull request message ✔️
- Well-organized and clean project ✔️
- Correct workflow was used ✔️
- Professional readme file ✔️
approv
# Status: Approved ✔️✔️✔️

Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100%.
Please, remember to tag me @uwadonat in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
```

**Pull request merged**